src/core/EdFi.DataManagementService.Core/ApiService.cs

```diff
@@ -40,6 +40,7 @@ ILogger<ApiService> _logger
                         new ApiSchemaValidationMiddleware(_apiSchemaProvider, _apiSchemaValidator, _logger),
                         new ProvideApiSchemaMiddleware(_apiSchemaProvider, _logger),
                         new ParsePathMiddleware(_logger),
+                        new ParseBodyMiddleware(_logger),
```

Comment:
This comes before `ValidateDocumentMiddleware`. I just issued this request: `curl -X POST http://localhost:5198/data/ed-fi/students -d "{notjson]"`. I expected an error coming from `ParseBodyMiddleWare`, but note the stack trace below:
 
```none
2024-05-24 16:17:16.453 ERR Unknown Error - TraceId { Value = 0HN3S6RHI5T5L:00000001 }System.Text.Json.JsonReaderException: 'n' is an invalid start of a property name. Expected a '"'. LineNumber: 0 | BytePositionInLine: 1.
   at System.Text.Json.ThrowHelper.ThrowJsonReaderException(Utf8JsonReader& json, ExceptionResource resource, Byte nextByte, ReadOnlySpan`1 bytes)
   at System.Text.Json.Utf8JsonReader.ReadSingleSegment()
   at System.Text.Json.Utf8JsonReader.Read()
   at System.Text.Json.JsonDocument.Parse(ReadOnlySpan`1 utf8JsonSpan, JsonReaderOptions readerOptions, MetadataDb& database, StackRowStack& stack)
   at System.Text.Json.JsonDocument.ParseUnrented(ReadOnlyMemory`1 utf8Json, JsonReaderOptions readerOptions, JsonTokenType tokenType)
   at System.Text.Json.JsonDocument.ParseValue(ReadOnlyMemory`1 json, JsonDocumentOptions options)
   at System.Text.Json.Nodes.JsonNode.Parse(String json, Nullable`1 nodeOptions, JsonDocumentOptions documentOptions)
   at EdFi.DataManagementService.Core.Middleware.ParseBodyMiddleware.Execute(PipelineContext context, Func`1 next) in D:\tanager\Data-Management-Service\src\core\EdFi.DataManagementService.Core\Middleware\ParseBodyMiddleware.cs:line 23

2024-05-24 16:17:16.473 DBG Entering ValidateEndpointMiddleware - TraceId { Value = 0HN3S6RHI5T5L:00000001 }
2024-05-24 16:17:16.480 DBG Entering ValidateDocumentMiddleware- TraceId { Value = 0HN3S6RHI5T5L:00000001 }
```

So the error was detected in the new middle ware, but we keep processing and the client response ends up coming from the validation middleware instead of from the `ParseBodyMiddleware`.